waitress/server.py

```diff
@@ -42,27 +42,95 @@ def create_server(application,
             'to return a WSGI app within your application.'
             )
     adj = Adjustments(**kw)
+
+    if map is None: # pragma: nocover
+        map = {}
+
+    dispatcher = _dispatcher
+    if dispatcher is None:
+        dispatcher = ThreadedTaskDispatcher()
+        dispatcher.set_thread_count(adj.threads)
+
     if adj.unix_socket and hasattr(socket, 'AF_UNIX'):
-        cls = UnixWSGIServer
-    else:
-        cls = TcpWSGIServer
-    return cls(application, map, _start, _sock, _dispatcher, adj)
+        sockinfo = (socket.AF_UNIX, socket.SOCK_STREAM, None, None)
+        return UnixWSGIServer(
+            application,
+            map,
+            _start,
+            _sock,
+            dispatcher=dispatcher,
+            adj=adj,
+            sockinfo=sockinfo)
+
+    effective_listen = []
+    last_serv = None
+    for sockinfo in adj.listen:
+        last_serv = TcpWSGIServer(
+            application,
+            map,
+            _start,
+            _sock,
+            dispatcher=dispatcher,
+            adj=adj,
+            sockinfo=sockinfo)
+        effective_listen.append((last_serv.effective_host, last_serv.effective_port))
+
+    # We are running a single server, so we can just return it...
+    if len(adj.listen) == 1:
+        return last_serv
+
+    return BaseServer(map, adj, effective_listen, dispatcher)
+
+class BaseServer(object):
```

Comment:
Another option would be to pass the various WSGI server instance objects into the multiserver class so it least appears that they aren't being constructed and then thrown away.  You'd still need to explain that they were jammed into the map, however, I guess.